src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,45 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_RANGE = 10;
+    private K[] keysRange;
+    private V[] valuesRange;
+    private int index;
+
+    public StorageImpl() {
+        keysRange = (K[]) new Object[MAX_RANGE];
+        valuesRange = (V[]) new Object[MAX_RANGE];
+        index = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int i = 0;
+        for (; i < index; i++) {
+            if (keysRange[i] == key || (keysRange[i] != null && keysRange[i].equals(key))) {
+                valuesRange[i] = value;
```

Comment:
```suggestion
        for (; i < index; i++) {
            if (compareKeys(keysRange[i], key) {
                valuesRange[i] = value;
      ```

you can make this equals in different metod

private boolean compareKeys(K key1, K key2) {
        return key1 == key2 || (key1 != null && key1.equals(key2));
    }

Reply:
you can make this equals in different metod

private boolean compareKeys(K key1, K key2) {
        return key1 == key2 || (key1 != null && key1.equals(key2));
    }